src/nlp/api.jl

```diff
@@ -1298,6 +1299,41 @@ function hess_op!(
   return LinearOperator{T}(nlp.meta.nvar, nlp.meta.nvar, true, true, prod!, prod!, prod!)
 end
 
+"""
+    P = tensor_projection(nlp, n, x, directions, args...)
+
+Returns the projection of the n-th derivative of the objective of `nlp` at `x` along the specified directions.
+
+#### Input arguments
+
+- `nlp`: An NLP model;
+- `n`: The order of the derivative to compute;
+- `x`: The point at which the derivative is evaluated;
+- `directions`: A tuple of indices specifying the directions (e.g., `(1, 2)` for a tensor projection along the first and second axes);
+- `args...`: A list of vectors, one for each direction specified in `directions`.
+"""
+function tensor_projection(
+    nlp::AbstractNLPModel{T, S},
+    n::Int,
+    x::AbstractVector,
+    directions::Tuple{Int, Vararg{Int}},
+    args...
+) where {T, S}
+    @lencheck nlp.meta.nvar x
+    m = n - length(directions)
+    @assert m ≥ 1
```

Comment:
Can we use NLPModels.@rangecheck instead of @assert ?

Reply:
Can we use `rangecheck` to ensure inequalities?

Reply:
Not tested, but I would expect `@rangecheck 1 typemax(Int) m` to work